ai_server/server.py

```diff
@@ -1,12 +1,250 @@
 from flask import Flask, request, jsonify, abort
 import ollama
+import subprocess
+import os
+import requests
+from typing import Optional
+from dotenv import load_dotenv
 
 from .redis_helper import REDIS_CONNECTION
 
+# Load environment variables from .env file
+load_dotenv()
+
 app = Flask('AI server')
 
 DEFAULT_MODEL = 'deepseek-coder-v2:latest'
 
+# Llama.cpp configuration
+LLAMA_CPP_CLI = '/data1/llama.cpp/bin/llama-cli'
+GGUF_DIR = '/data1/GGUF'
+
+# Llama server configuration - now uses environment variable
+LLAMA_SERVER_URL = os.getenv('LLAMA_SERVER_URL')  # e.g., http://localhost:8080
```

Comment:
This is slightly different from @wkukka1's interpretation, which doesn't include the `http://`. Let's include it in both.